bootloader/grub.go

```diff
@@ -80,24 +79,34 @@ func (g *grub) dir() string {
 	return filepath.Join(g.rootdir, g.basedir)
 }
 
-func (g *grub) installRecoveryBootConfig(gadgetDir string) (bool, error) {
+func (g *grub) installManagedRecoveryBootConfig(gadgetDir string) (bool, error) {
 	gadgetGrubCfg := filepath.Join(gadgetDir, g.Name()+".conf")
 	if !osutil.FileExists(gadgetGrubCfg) {
 		// gadget does not use grub bootloader
 		return false, nil
 	}
 	assetName := g.Name() + "-recovery.cfg"
-	bootConfig := assets.Internal(assetName)
-	if bootConfig == nil {
-		return true, fmt.Errorf("internal error: no boot asset for %q", assetName)
+	systemFile := filepath.Join(g.rootdir, "/EFI/ubuntu/grub.cfg")
+	return genericSetBootConfigFromAsset(systemFile, assetName)
+}
+
+func (g *grub) installManagedBootConfig(gadgetDir string) (bool, error) {
+	gadgetGrubCfg := filepath.Join(gadgetDir, g.Name()+".conf")
+	if !osutil.FileExists(gadgetGrubCfg) {
+		// gadget does not use grub bootloader
+		return false, nil
 	}
+	assetName := g.Name() + ".cfg"
 	systemFile := filepath.Join(g.rootdir, "/EFI/ubuntu/grub.cfg")
-	return genericSetBootConfig(systemFile, bootConfig)
+	return genericSetBootConfigFromAsset(systemFile, assetName)
```

Comment:
we don't seem to have a test in this package for this

Reply:
Yeah, there were only some indirect tests in `boot`. Extended the local unit tests to cover this.